src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -74,6 +79,42 @@ def mixins(self):
   def _setup_command(self):
     return self.__setup_command
 
+  def run(self):
```

Comment:
I'm sceptical of the need to modify this class as noted below, but assuming it is somehow actually needed - this class needs to move to a shared file. >50% of its code is not needed locally.